WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardView.swift

```diff
@@ -71,6 +72,10 @@ private extension BlazeCampaignDashboardViewHostingController {
     /// Handles navigation to the campaign creation view.
     func navigateToCampaignCreation(source: BlazeSource, productID: Int64? = nil) {
         if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.blazei3NativeCampaignCreation) {
+            guard let productID else {
+                return navigateToBlazeProductSelector(source: source)
+            }
```

Comment:
We were checking for `viewModel.shouldShowProductSelectorView` below before opening the product selector. I think the logic needs to be updated:
```suggestion
            if let productID {
                navigateToNativeCampaignCreation(source: source, productID: productID)
            } else if viewModel.shouldShowProductSelectorView {
                navigateToBlazeProductSelector(source: source)
            } else {
                // TODO: get the single product ID from the view model and navigate to campaign creation
            }
```

Reply:
Updated in 5a1d3e6e00803dfebf9f23e63fa9ac94a63b73ac and 7bc96285998d1a8b4f2b435ce1c3130623444905